app/controllers/api/base_controller.rb

```diff
@@ -36,6 +36,11 @@ class BaseController < ApplicationController
                          :only => [:show, :update, :destroy, :handle_options_request, :options]
     end
     skip_before_action :get_global_session_data
+    skip_before_action :reset_toolbar
+    skip_before_action :set_session_tenant
+    skip_before_action :set_user_time_zone
+    skip_before_action :set_gettext_locale
```

Comment:
Hi @imtayadeway I don't think set_gettext_locale should be skipped as it sets the FastGettext locale for User.current_user, so if strings are returned let's say in /api entrypoint with I18n.t and latest gettext, we'd lose the locale setting for the user.